### PR TITLE
Bundle Adjustment using RobustCostFunction

### PR DESCRIPTION
**Pull request by luisenp**
Serves as a test for #148. From log trace below it seems to be working, since both inner optimizer error decreases and outer loss goes down.

```bash
2022-04-08 04:31:07,008][__main__][INFO] - CAMERA LOSS (no learning):   1.722
[2022-04-08 04:31:07,008][__main__][INFO] - Input histogram:
[2022-04-08 04:31:07,256][__main__][INFO] - 0-1: ################################# 186.0
[2022-04-08 04:31:07,256][__main__][INFO] - 1-2: ################################################################################ 448.0
[2022-04-08 04:31:07,256][__main__][INFO] - 2-3: ############################################################################ 428.0
[2022-04-08 04:31:07,258][__main__][INFO] - 3-4: ############################################ 248.0
[2022-04-08 04:31:07,259][__main__][INFO] - 4-5: ############################# 162.0
[2022-04-08 04:31:07,260][__main__][INFO] - 5-6: ######### 53.0
[2022-04-08 04:31:07,261][__main__][INFO] - 6-7: ### 19.0
[2022-04-08 04:31:07,262][__main__][INFO] - 7-8: ## 10.0
[2022-04-08 04:31:07,263][__main__][INFO] - 8-9:  2.0
[2022-04-08 04:31:07,264][__main__][INFO] - 9-10:  1.0
[2022-04-08 04:31:07,265][__main__][INFO] - 10+: ######## 43.0
[2022-04-08 04:31:07,266][__main__][INFO] - 
[2022-04-08 04:31:07,268][__main__][INFO] -  ******************* EPOCH 0 ******************* 
Nonlinear optimizer. Iteration: 0. Error: 16518.708089590502
Nonlinear optimizer. Iteration: 1. Error: 15431.338892805641
Nonlinear optimizer. Iteration: 2. Error: 14529.230912020023
Nonlinear optimizer. Iteration: 3. Error: 13785.035380732541
Nonlinear optimizer. Iteration: 4. Error: 13174.34906106902
Nonlinear optimizer. Iteration: 5. Error: 12675.13033086467
Nonlinear optimizer. Iteration: 6. Error: 12268.013144886916
Nonlinear optimizer. Iteration: 7. Error: 11936.813690696694
Nonlinear optimizer. Iteration: 8. Error: 11667.73334755549
Nonlinear optimizer. Iteration: 9. Error: 11449.372022225027
Nonlinear optimizer. Iteration: 1. Error: 11272.315996603407
[2022-04-08 04:31:20,560][__main__][INFO] - Output histogram:
[2022-04-08 04:31:20,905][__main__][INFO] - 0-1: ############################################################# 584.0
[2022-04-08 04:31:20,908][__main__][INFO] - 1-2: ################################################################################ 761.0
[2022-04-08 04:31:20,910][__main__][INFO] - 2-3: ##################### 198.0
[2022-04-08 04:31:20,914][__main__][INFO] - 3-4: # 14.0
[2022-04-08 04:31:20,916][__main__][INFO] - 4-5:  0.0
[2022-04-08 04:31:20,917][__main__][INFO] - 5-6:  0.0
[2022-04-08 04:31:20,919][__main__][INFO] - 6-7:  0.0
[2022-04-08 04:31:20,921][__main__][INFO] - 7-8:  0.0
[2022-04-08 04:31:20,922][__main__][INFO] - 8-9:  0.0
[2022-04-08 04:31:20,924][__main__][INFO] - 9-10:  0.0
[2022-04-08 04:31:20,925][__main__][INFO] - 10+: ##### 43.0
[2022-04-08 04:31:20,926][__main__][INFO] - 
[2022-04-08 04:31:20,927][__main__][INFO] - Epoch: 0 Loss: -0.5888410137276036 Kernel Radius: exp(3.0999973109538357)=22.19789159020614
[2022-04-08 04:31:20,928][__main__][INFO] - Epoch took  13.281 seconds
[2022-04-08 04:31:20,939][__main__][INFO] -  ******************* EPOCH 1 ******************* 
Nonlinear optimizer. Iteration: 0. Error: 17018.64156205428
Nonlinear optimizer. Iteration: 1. Error: 15916.156256384418
Nonlinear optimizer. Iteration: 2. Error: 15004.760700580819
Nonlinear optimizer. Iteration: 3. Error: 14255.268097557928
Nonlinear optimizer. Iteration: 4. Error: 13641.74695166519
Nonlinear optimizer. Iteration: 5. Error: 13140.940578561655
Nonlinear optimizer. Iteration: 6. Error: 12733.155735866623
Nonlinear optimizer. Iteration: 7. Error: 12401.80716347349
Nonlinear optimizer. Iteration: 8. Error: 12132.803447667282
Nonlinear optimizer. Iteration: 9. Error: 11914.653752875238
Nonlinear optimizer. Iteration: 1. Error: 11737.75078468002
[2022-04-08 04:31:34,209][__main__][INFO] - Output histogram:
[2022-04-08 04:31:34,553][__main__][INFO] - 0-1: ############################################################# 582.0
[2022-04-08 04:31:34,556][__main__][INFO] - 1-2: ################################################################################ 763.0
[2022-04-08 04:31:34,557][__main__][INFO] - 2-3: ##################### 198.0
[2022-04-08 04:31:34,559][__main__][INFO] - 3-4: # 14.0
[2022-04-08 04:31:34,560][__main__][INFO] - 4-5:  0.0
[2022-04-08 04:31:34,561][__main__][INFO] - 5-6:  0.0
[2022-04-08 04:31:34,563][__main__][INFO] - 6-7:  0.0
[2022-04-08 04:31:34,564][__main__][INFO] - 7-8:  0.0
[2022-04-08 04:31:34,565][__main__][INFO] - 8-9:  0.0
[2022-04-08 04:31:34,566][__main__][INFO] - 9-10:  0.0
[2022-04-08 04:31:34,567][__main__][INFO] - 10+: ##### 43.0
[2022-04-08 04:31:34,568][__main__][INFO] - 
[2022-04-08 04:31:34,569][__main__][INFO] - Epoch: 1 Loss: -0.5895888960289959 Kernel Radius: exp(3.200112765251785)=24.53529677003766
[2022-04-08 04:31:34,570][__main__][INFO] - Epoch took  13.269 seconds
[2022-04-08 04:31:34,583][__main__][INFO] -  ******************* EPOCH 2 ******************* 
Nonlinear optimizer. Iteration: 0. Error: 17535.73703814031
Nonlinear optimizer. Iteration: 1. Error: 16420.211180076964
Nonlinear optimizer. Iteration: 2. Error: 15500.899769138214
Nonlinear optimizer. Iteration: 3. Error: 14747.023264479103
Nonlinear optimizer. Iteration: 4. Error: 14130.985417928548
Nonlinear optimizer. Iteration: 5. Error: 13628.856170475197
Nonlinear optimizer. Iteration: 6. Error: 13220.654966783732
Nonlinear optimizer. Iteration: 7. Error: 12889.167047201765
Nonlinear optimizer. Iteration: 8. Error: 12620.26466308786
Nonlinear optimizer. Iteration: 9. Error: 12402.204594164785
Nonlinear optimizer. Iteration: 1. Error: 12225.353392872858
[2022-04-08 04:31:47,862][__main__][INFO] - Output histogram:
[2022-04-08 04:31:48,207][__main__][INFO] - 0-1: ############################################################# 581.0
[2022-04-08 04:31:48,210][__main__][INFO] - 1-2: ################################################################################ 764.0
[2022-04-08 04:31:48,211][__main__][INFO] - 2-3: ##################### 199.0
[2022-04-08 04:31:48,212][__main__][INFO] - 3-4: # 13.0
[2022-04-08 04:31:48,217][__main__][INFO] - 4-5:  0.0
[2022-04-08 04:31:48,218][__main__][INFO] - 5-6:  0.0
[2022-04-08 04:31:48,219][__main__][INFO] - 6-7:  0.0
[2022-04-08 04:31:48,220][__main__][INFO] - 7-8:  0.0
[2022-04-08 04:31:48,221][__main__][INFO] - 8-9:  0.0
[2022-04-08 04:31:48,223][__main__][INFO] - 9-10:  0.0
[2022-04-08 04:31:48,223][__main__][INFO] - 10+: ##### 43.0
[2022-04-08 04:31:48,224][__main__][INFO] - 
[2022-04-08 04:31:48,224][__main__][INFO] - Epoch: 2 Loss: -0.5902865946292938 Kernel Radius: exp(3.300414893462347)=27.123890111154342
[2022-04-08 04:31:48,231][__main__][INFO] - Epoch took  13.278 seconds
[2022-04-08 04:31:48,243][__main__][INFO] -  ******************* EPOCH 3 ******************* 
Nonlinear optimizer. Iteration: 0. Error: 18071.99042796631
Nonlinear optimizer. Iteration: 1. Error: 16945.354632730443
Nonlinear optimizer. Iteration: 2. Error: 16019.42237783891
Nonlinear optimizer. Iteration: 3. Error: 15261.736381995182
Nonlinear optimizer. Iteration: 4. Error: 14643.521156546076
Nonlinear optimizer. Iteration: 5. Error: 14140.462146435078
Nonlinear optimizer. Iteration: 6. Error: 13731.861396976767
Nonlinear optimizer. Iteration: 7. Error: 13400.258874131341
Nonlinear optimizer. Iteration: 8. Error: 13131.34640213992
Nonlinear optimizer. Iteration: 9. Error: 12913.251726462378
Nonlinear optimizer. Iteration: 1. Error: 12736.34882418398
[2022-04-08 04:32:01,421][__main__][INFO] - Output histogram:
[2022-04-08 04:32:01,764][__main__][INFO] - 0-1: ############################################################# 581.0
[2022-04-08 04:32:01,768][__main__][INFO] - 1-2: ################################################################################ 764.0
[2022-04-08 04:32:01,771][__main__][INFO] - 2-3: ##################### 200.0
[2022-04-08 04:32:01,773][__main__][INFO] - 3-4: # 12.0
[2022-04-08 04:32:01,775][__main__][INFO] - 4-5:  0.0
[2022-04-08 04:32:01,776][__main__][INFO] - 5-6:  0.0
[2022-04-08 04:32:01,777][__main__][INFO] - 6-7:  0.0
[2022-04-08 04:32:01,778][__main__][INFO] - 7-8:  0.0
[2022-04-08 04:32:01,779][__main__][INFO] - 8-9:  0.0
[2022-04-08 04:32:01,781][__main__][INFO] - 9-10:  0.0
[2022-04-08 04:32:01,783][__main__][INFO] - 10+: ##### 43.0
[2022-04-08 04:32:01,784][__main__][INFO] - 
[2022-04-08 04:32:01,786][__main__][INFO] - Epoch: 3 Loss: -0.5908859703477123 Kernel Radius: exp(3.400961144021781)=29.99291370784619
[2022-04-08 04:32:01,787][__main__][INFO] - Epoch took  13.178 seconds
[2022-04-08 04:32:01,796][__main__][INFO] -  ******************* EPOCH 4 ******************* 
Nonlinear optimizer. Iteration: 0. Error: 18629.332154644253
Nonlinear optimizer. Iteration: 1. Error: 17493.049966685452
Nonlinear optimizer. Iteration: 2. Error: 16561.506749760396
Nonlinear optimizer. Iteration: 3. Error: 15800.51968393859
Nonlinear optimizer. Iteration: 4. Error: 15180.570489113503
Nonlinear optimizer. Iteration: 5. Error: 14676.83755101105
Nonlinear optimizer. Iteration: 6. Error: 14267.86642214105
Nonlinear optimizer. Iteration: 7. Error: 13936.148001296662
Nonlinear optimizer. Iteration: 8. Error: 13667.113919818265
Nonlinear optimizer. Iteration: 9. Error: 13448.89082827833
Nonlinear optimizer. Iteration: 1. Error: 13271.857115526533
[2022-04-08 04:32:15,050][__main__][INFO] - Output histogram:
[2022-04-08 04:32:15,393][__main__][INFO] - 0-1: ############################################################# 581.0
[2022-04-08 04:32:15,396][__main__][INFO] - 1-2: ################################################################################ 764.0
[2022-04-08 04:32:15,397][__main__][INFO] - 2-3: ##################### 200.0
[2022-04-08 04:32:15,398][__main__][INFO] - 3-4: # 12.0
[2022-04-08 04:32:15,399][__main__][INFO] - 4-5:  0.0
[2022-04-08 04:32:15,400][__main__][INFO] - 5-6:  0.0
[2022-04-08 04:32:15,401][__main__][INFO] - 6-7:  0.0
[2022-04-08 04:32:15,404][__main__][INFO] - 7-8:  0.0
[2022-04-08 04:32:15,404][__main__][INFO] - 8-9:  0.0
[2022-04-08 04:32:15,404][__main__][INFO] - 9-10:  0.0
[2022-04-08 04:32:15,404][__main__][INFO] - 10+: ##### 43.0
[2022-04-08 04:32:15,404][__main__][INFO] - 
[2022-04-08 04:32:15,404][__main__][INFO] - Epoch: 4 Loss: -0.5914315425830889 Kernel Radius: exp(3.5017931970635323)=33.174887764031396
[2022-04-08 04:32:15,404][__main__][INFO] - Epoch took  13.253 seconds
[2022-04-08 04:32:15,412][__main__][INFO] -  ******************* EPOCH 5 ******************* 
Nonlinear optimizer. Iteration: 0. Error: 19209.636670980548
Nonlinear optimizer. Iteration: 1. Error: 18065.037638331793
Nonlinear optimizer. Iteration: 2. Error: 17128.606748950137
Nonlinear optimizer. Iteration: 3. Error: 16364.845462632176
Nonlinear optimizer. Iteration: 4. Error: 15743.700673802688
Nonlinear optimizer. Iteration: 5. Error: 15239.275977320216
Nonlinear optimizer. Iteration: 6. Error: 14829.981606298687
Nonlinear optimizer. Iteration: 7. Error: 14498.027118832882
Nonlinear optimizer. Iteration: 8. Error: 14228.763905251486
Nonlinear optimizer. Iteration: 9. Error: 14010.321264771175
Nonlinear optimizer. Iteration: 1. Error: 13833.079292428532
[2022-04-08 04:32:28,592][__main__][INFO] - Output histogram:
[2022-04-08 04:32:28,928][__main__][INFO] - 0-1: ############################################################# 581.0
[2022-04-08 04:32:28,931][__main__][INFO] - 1-2: ################################################################################ 765.0
[2022-04-08 04:32:28,936][__main__][INFO] - 2-3: ##################### 199.0
[2022-04-08 04:32:28,937][__main__][INFO] - 3-4: # 12.0
[2022-04-08 04:32:28,937][__main__][INFO] - 4-5:  0.0
[2022-04-08 04:32:28,938][__main__][INFO] - 5-6:  0.0
[2022-04-08 04:32:28,940][__main__][INFO] - 6-7:  0.0
[2022-04-08 04:32:28,941][__main__][INFO] - 7-8:  0.0
[2022-04-08 04:32:28,943][__main__][INFO] - 8-9:  0.0
[2022-04-08 04:32:28,943][__main__][INFO] - 9-10:  0.0
[2022-04-08 04:32:28,944][__main__][INFO] - 10+: #### 43.0
[2022-04-08 04:32:28,945][__main__][INFO] - 
[2022-04-08 04:32:28,951][__main__][INFO] - Epoch: 5 Loss: -0.5919399846198976 Kernel Radius: exp(3.602933643186478)=36.705758245910886
[2022-04-08 04:32:28,955][__main__][INFO] - Epoch took  13.180 seconds
[2022-04-08 04:32:28,967][__main__][INFO] -  ******************* EPOCH 6 ******************* 
Nonlinear optimizer. Iteration: 0. Error: 19814.429318758535
Nonlinear optimizer. Iteration: 1. Error: 18662.729303770575
Nonlinear optimizer. Iteration: 2. Error: 17722.113677299465
Nonlinear optimizer. Iteration: 3. Error: 16956.245210547615
Nonlinear optimizer. Iteration: 4. Error: 16334.047259098308
Nonlinear optimizer. Iteration: 5. Error: 15828.996513807606
Nonlinear optimizer. Iteration: 6. Error: 15419.323091781831
Nonlinear optimizer. Iteration: 7. Error: 15087.013491737254
Nonlinear optimizer. Iteration: 8. Error: 14817.419601719357
Nonlinear optimizer. Iteration: 9. Error: 14598.670561293657
Nonlinear optimizer. Iteration: 1. Error: 14421.145842256165
[2022-04-08 04:32:42,469][__main__][INFO] - Output histogram:
[2022-04-08 04:32:42,820][__main__][INFO] - 0-1: ############################################################# 581.0
[2022-04-08 04:32:42,823][__main__][INFO] - 1-2: ################################################################################ 765.0
[2022-04-08 04:32:42,826][__main__][INFO] - 2-3: ##################### 199.0
[2022-04-08 04:32:42,828][__main__][INFO] - 3-4: # 12.0
[2022-04-08 04:32:42,829][__main__][INFO] - 4-5:  0.0
[2022-04-08 04:32:42,831][__main__][INFO] - 5-6:  0.0
[2022-04-08 04:32:42,832][__main__][INFO] - 6-7:  0.0
[2022-04-08 04:32:42,834][__main__][INFO] - 7-8:  0.0
[2022-04-08 04:32:42,836][__main__][INFO] - 8-9:  0.0
[2022-04-08 04:32:42,837][__main__][INFO] - 9-10:  0.0
[2022-04-08 04:32:42,838][__main__][INFO] - 10+: #### 43.0
[2022-04-08 04:32:42,839][__main__][INFO] - 
[2022-04-08 04:32:42,842][__main__][INFO] - Epoch: 6 Loss: -0.5924268540324825 Kernel Radius: exp(3.7043716059916796)=40.62451109495164
[2022-04-08 04:32:42,843][__main__][INFO] - Epoch took  13.502 seconds
[2022-04-08 04:32:42,853][__main__][INFO] -  ******************* EPOCH 7 ******************* 
Nonlinear optimizer. Iteration: 0. Error: 20445.274064271092
Nonlinear optimizer. Iteration: 1. Error: 19287.420318133976
Nonlinear optimizer. Iteration: 2. Error: 18343.41551246025
Nonlinear optimizer. Iteration: 3. Error: 17575.911618346836
Nonlinear optimizer. Iteration: 4. Error: 16952.69117120948
Nonlinear optimizer. Iteration: 5. Error: 16447.023844412608
Nonlinear optimizer. Iteration: 6. Error: 16036.8402543105
Nonlinear optimizer. Iteration: 7. Error: 15704.062728915324
Nonlinear optimizer. Iteration: 8. Error: 15434.04084408201
Nonlinear optimizer. Iteration: 9. Error: 15214.901232781394
Nonlinear optimizer. Iteration: 1. Error: 15037.020779731522
[2022-04-08 04:32:56,418][__main__][INFO] - Output histogram:
[2022-04-08 04:32:56,750][__main__][INFO] - 0-1: ############################################################ 579.0
[2022-04-08 04:32:56,753][__main__][INFO] - 1-2: ################################################################################ 767.0
[2022-04-08 04:32:56,756][__main__][INFO] - 2-3: ##################### 199.0
[2022-04-08 04:32:56,757][__main__][INFO] - 3-4: # 12.0
[2022-04-08 04:32:56,759][__main__][INFO] - 4-5:  0.0
[2022-04-08 04:32:56,760][__main__][INFO] - 5-6:  0.0
[2022-04-08 04:32:56,761][__main__][INFO] - 6-7:  0.0
[2022-04-08 04:32:56,761][__main__][INFO] - 7-8:  0.0
[2022-04-08 04:32:56,762][__main__][INFO] - 8-9:  0.0
[2022-04-08 04:32:56,763][__main__][INFO] - 9-10:  0.0
[2022-04-08 04:32:56,765][__main__][INFO] - 10+: #### 43.0
[2022-04-08 04:32:56,766][__main__][INFO] - 
[2022-04-08 04:32:56,768][__main__][INFO] - Epoch: 7 Loss: -0.5928866245543492 Kernel Radius: exp(3.8060417807229765)=44.972076759869964
[2022-04-08 04:32:56,780][__main__][INFO] - Epoch took  13.565 seconds
[2022-04-08 04:32:56,791][__main__][INFO] -  ******************* EPOCH 8 ******************* 
Nonlinear optimizer. Iteration: 0. Error: 21103.359285258775
Nonlinear optimizer. Iteration: 1. Error: 19940.262399259504
Nonlinear optimizer. Iteration: 2. Error: 18993.619307064768
Nonlinear optimizer. Iteration: 3. Error: 18224.591329489944
Nonlinear optimizer. Iteration: 4. Error: 17600.3964773766
Nonlinear optimizer. Iteration: 5. Error: 17094.014147818638
Nonlinear optimizer. Iteration: 6. Error: 16683.182010960045
Nonlinear optimizer. Iteration: 7. Error: 16349.817041105322
Nonlinear optimizer. Iteration: 8. Error: 16079.263665345228
Nonlinear optimizer. Iteration: 9. Error: 15859.643633637923
Nonlinear optimizer. Iteration: 1. Error: 15681.329248870446
[2022-04-08 04:33:09,691][__main__][INFO] - Output histogram:
[2022-04-08 04:33:10,061][__main__][INFO] - 0-1: ############################################################ 580.0
[2022-04-08 04:33:10,065][__main__][INFO] - 1-2: ################################################################################ 767.0
[2022-04-08 04:33:10,066][__main__][INFO] - 2-3: ##################### 198.0
[2022-04-08 04:33:10,067][__main__][INFO] - 3-4: # 12.0
[2022-04-08 04:33:10,068][__main__][INFO] - 4-5:  0.0
[2022-04-08 04:33:10,069][__main__][INFO] - 5-6:  0.0
[2022-04-08 04:33:10,070][__main__][INFO] - 6-7:  0.0
[2022-04-08 04:33:10,071][__main__][INFO] - 7-8:  0.0
[2022-04-08 04:33:10,072][__main__][INFO] - 8-9:  0.0
[2022-04-08 04:33:10,073][__main__][INFO] - 9-10:  0.0
[2022-04-08 04:33:10,075][__main__][INFO] - 10+: #### 43.0
[2022-04-08 04:33:10,076][__main__][INFO] - 
[2022-04-08 04:33:10,076][__main__][INFO] - Epoch: 8 Loss: -0.5933228945194194 Kernel Radius: exp(3.9077939052687367)=49.78899149457706
[2022-04-08 04:33:10,077][__main__][INFO] - Epoch took  12.901 seconds
[2022-04-08 04:33:10,087][__main__][INFO] -  ******************* EPOCH 9 ******************* 
Nonlinear optimizer. Iteration: 0. Error: 21789.182557856075
Nonlinear optimizer. Iteration: 1. Error: 20621.90974492445
Nonlinear optimizer. Iteration: 2. Error: 19672.96916919971
Nonlinear optimizer. Iteration: 3. Error: 18902.46523076101
Nonlinear optimizer. Iteration: 4. Error: 18277.240384561388
Nonlinear optimizer. Iteration: 5. Error: 17769.976591128852
Nonlinear optimizer. Iteration: 6. Error: 17358.35163486934
Nonlinear optimizer. Iteration: 7. Error: 17024.27386883125
Nonlinear optimizer. Iteration: 8. Error: 16753.07982427145
Nonlinear optimizer. Iteration: 9. Error: 16532.884054096823
Nonlinear optimizer. Iteration: 1. Error: 16354.052326098103
[2022-04-08 04:33:23,662][__main__][INFO] - Output histogram:
[2022-04-08 04:33:24,006][__main__][INFO] - 0-1: ############################################################# 581.0
[2022-04-08 04:33:24,009][__main__][INFO] - 1-2: ################################################################################ 767.0
[2022-04-08 04:33:24,016][__main__][INFO] - 2-3: ##################### 197.0
[2022-04-08 04:33:24,017][__main__][INFO] - 3-4: # 12.0
[2022-04-08 04:33:24,018][__main__][INFO] - 4-5:  0.0
[2022-04-08 04:33:24,019][__main__][INFO] - 5-6:  0.0
[2022-04-08 04:33:24,022][__main__][INFO] - 6-7:  0.0
[2022-04-08 04:33:24,023][__main__][INFO] - 7-8:  0.0
[2022-04-08 04:33:24,025][__main__][INFO] - 8-9:  0.0
[2022-04-08 04:33:24,027][__main__][INFO] - 9-10:  0.0
[2022-04-08 04:33:24,028][__main__][INFO] - 10+: #### 43.0
[2022-04-08 04:33:24,029][__main__][INFO] - 
[2022-04-08 04:33:24,031][__main__][INFO] - Epoch: 9 Loss: -0.5937097053399462 Kernel Radius: exp(4.009330793831091)=55.10997828313432
[2022-04-08 04:33:24,032][__main__][INFO] - Epoch took  13.574 seconds
```

### CODE REVIEW

**Comment by luisenp** [comment]
This should work after #148 and #205 are merged.

**Review by fantaosha — APPROVED** [review]
LGTM

File: theseus/utils/examples/bundle_adjustment/reprojection_error.py
Comment (mhmukadam) [inline theseus/utils/examples/bundle_adjustment/reprojection_error.py:13]: Maybe we can move `Reprojection` to embodied/measurements?

**Review by mhmukadam — APPROVED** [review]
Looks good. Maybe merge this after merging #213.

File: examples/bundle_adjustment.py
Comment (mhmukadam) [inline examples/bundle_adjustment.py:185]: We can also turn vectorize on now that it is in main.

File: theseus/utils/examples/bundle_adjustment/reprojection_error.py
Comment (luisenp) [reply theseus/utils/examples/bundle_adjustment/reprojection_error.py:13]: Done.

File: theseus/utils/examples/bundle_adjustment/tests/test_reprojection_error.py
Comment (mhmukadam) [inline theseus/utils/examples/bundle_adjustment/tests/test_reprojection_error.py:54]: You can move test file to embodied as well, right?

**Review by mhmukadam — APPROVED** [review]
Looks good. py37 CI is failing.

**Pull request merged**
